src/centroiders.cpp

```diff
@@ -105,9 +107,98 @@ int BasicThresholdOnePass(unsigned char *image, int imageWidth, int imageHeight)
+        // scheme will have 1 extra row/column, and these always start with the first rows/columns.
+    // div: The minimum rows/columns each box will have.
+    // For leftover and div, "horizontal" refers to rows and "vertical" refers to columns
+
+// By default, this algorithm ensures that regardless of the subdivisions entered, the number of
+// pixels in each subdivision is imageWidth*imageHeight <= size <= 100 to ensure that enough
+// pixles will be sampled in each subdivision
+
+// Accepts an index in the subdivision scheme, and a horizontal/vertical leftover and div
+// This method uses these quantities to find the first row/column index in the actual image
+// that "i" references.
+int StartOfSubdivision(int i, int leftover, int div) {
+    if(i < leftover) {
+        return i * (div + 1);
+    } else {
+        return leftover * (div + 1) + (i - leftover) * div;
+    }
+}
+
+// Refer to definition for details
+long FindSubdivision(long i, int imageWidth, int imageHeight, int subdivisions);
+
+
+// Accepts a subdivsion, or "box" number, the image's brightness array and associated dimensions, 
+// and the subdivisions.
+// Returns the threshold for the corresponding box.
+int LocalBasicThreshold(int box, unsigned char *image, int imageWidth, int imageHeight, 
+        int subdivisions) {
+
+    int horizontalDiv = imageHeight / subdivisions;
+    int verticalDiv = imageWidth / subdivisions;
+    int horizontalLeftover = imageHeight % subdivisions;
+    int verticalLeftover = imageWidth % subdivisions;
+    int row = box / subdivisions; // Finds the current row in the subdivision scheme
+    int col = box % subdivisions; // Finds the current column in the subdivision scheme
+    double average = 0;
+    double squareSum = 0;
+    long count = 0;
+
+    // Runs through "box" in row-major order
+    for(int i = StartOfSubdivision(row, horizontalLeftover, horizontalDiv); 
+            i < StartOfSubdivision(row + 1, horizontalLeftover, horizontalDiv); i++) {
+        for(int j = StartOfSubdivision(col, verticalLeftover, verticalDiv); 
+                j < StartOfSubdivision(col + 1, verticalLeftover, verticalDiv); j++) {
+            average += image[i * imageWidth + j];
+            squareSum += image[i * imageWidth + j] * image[i * imageWidth + j];
+            count++;
+
+            // Checks for Error
+            assert(FindSubdivision(i*imageWidth+j, imageWidth, imageHeight, subdivisions) == box);
+        }
+    }
+
+    average /= count;
+    return std::round(average + (5 * std::sqrt((squareSum - count * average * average) / (count - 1))));
+}
+
+// Accepts the image's brightness array and dimensions, and the subdivisions, and returns
+// a vector with the threshold for each "box" in row-major order
+std::vector<int> LocalThresholding(unsigned char *image, int imageWidth, int imageHeight, 
+        int subdivisions) {
+    
+    std::vector<int> standardDeviations;
```

Comment:
Variable name is misleading; it doesn't store standard deviations, it stores thresholds